src/app/model-actions/model-actions.component.ts

```diff
@@ -0,0 +1,100 @@
+import {Component, OnInit} from '@angular/core';
+import {FormArray, FormBuilder, FormControl, FormGroup, Validators} from '@angular/forms';
+import {EntityActionsComponent} from '../entity-actions/entity-actions.component';
+import {ActivatedRoute, Params} from '@angular/router';
+import {ProjectService} from '../../services/project-service/project.service';
+import {ToastrService} from 'ngx-toastr';
+import {zip} from 'rxjs';
+import {Model} from '../../interfaces/model';
+import {ModelService} from '../../services/model-service/model.service';
+import {Project} from '../../interfaces/project';
+import {EntitiesEnum} from '../../enums/entities.enum';
+
+@Component({
+  selector: 'app-model-actions',
+  templateUrl: './model-actions.component.html',
+  styleUrls: ['./model-actions.component.css']
+})
+export class ModelActionsComponent extends EntityActionsComponent implements OnInit {
+  modelForm: FormGroup;
+  public project: Project;
+  public numberFields: number;
+  public types;
+
+  constructor(private formBuilder: FormBuilder, private activatedRoute: ActivatedRoute,
+              private projectService: ProjectService, private modelService: ModelService,
+              protected toastr: ToastrService) {
+    super(toastr);
+    this.types = modelService.types;
+  }
+
+  ngOnInit(): void {
+    const mergeObservables = zip(this.activatedRoute.queryParams, this.projectService.getProjects());
```

Comment:
You can add typing and use destructuring to unwrap params and projects in different parameters:
```typescript
    const mergeObservables = zip<[Params, Project[]]>(this.activatedRoute.queryParams, this.projectService.getProjects());
    mergeObservables.subscribe(([params, projects]) => {
      this.project = projects.find(project => project.name === params.projectName);
      this.name = params.modelName;
      this.edit = params.edit;
```